etc/make_small_lib.pl

```diff
@@ -0,0 +1,407 @@
+#!/usr/bin/env perl
+
+use strict;
+use warnings;
+
+use Getopt::Long;
+use File::Basename 'basename';
+use File::Glob 'bsd_glob';
+use File::Spec;
+use List::MoreUtils 'any';
+
+# OS dependent path separator
+my $sep = File::Spec->catfile('', '');
+
+# A list of the functions that trade space for speed.
+# They will be removed from the list if the option "-n" is given
+my %fast_functions = (
+       s_mp_balance_mul            => 1,
+       s_mp_exptmod_fast           => 1,
+       s_mp_invmod_fast            => 1,
+       s_mp_karatsuba_mul          => 1,
+       s_mp_karatsuba_sqr          => 1,
+       s_mp_montgomery_reduce_fast => 1,
+       s_mp_mul_digs_fast          => 1,
+       s_mp_mul_high_digs_fast     => 1,
+       s_mp_sqr_fast               => 1,
+       s_mp_toom_mul               => 1,
+       s_mp_toom_sqr               => 1
+   );
+
+my %deprecated_functions = (
+       mp_get_bit                  => 's_mp_get_bit',
+       mp_jacobi                   => 'mp_kronecker',
+       mp_prime_random_ex          => 's_mp_prime_random_ex',
+       mp_rand_digit               => 's_mp_rand_source',
+       fast_mp_invmod              => 's_mp_invmod_fast',
+       fast_mp_montgomery_reduce   => 's_mp_montgomery_reduce_fast',
+       fast_s_mp_mul_digs          => 's_mp_mul_digs_fast',
+       fast_s_mp_mul_high_digs     => 's_mp_mul_high_digs_fast',
+       fast_s_mp_sqr               => 's_mp_sqr_fast',
+       mp_balance_mul              => 's_mp_balance_mul',
+       mp_exptmod_fast             => 's_mp_exptmod_fast',
+       mp_invmod_slow              => 's_mp_invmod_slow',
+       mp_karatsuba_mul            => 's_mp_karatsuba_mul',
+       mp_karatsuba_sqr            => 's_mp_karatsuba_sqr',
+       mp_toom_mul                 => 's_mp_toom_mul',
+       mp_toom_sqr                 => 's_mp_toom_sqr',
+       bn_reverse                  => 's_mp_reverse',
+       mp_tc_and                   => 'mp_and',
+       mp_tc_or                    => 'mp_or',
+       mp_tc_xor                   => 'mp_xor',
+       mp_tc_div_2d                => 'mp_signed_rsh',
+       mp_init_set_int             => 'mp_init_u32',
+       mp_set_int                  => 'mp_set_u32',
+       mp_set_long                 => 'mp_set_u64',
+       mp_set_long_long            => 'mp_set_u64',
+       mp_get_int                  => 'mp_get_mag32',
+       mp_get_long_long            => 'mp_get_mag64',
+       mp_prime_is_divisible       => 's_mp_prime_is_divisible',
+       mp_expt_d_ex                => 'mp_expt_d',
+       mp_n_root_ex                => 'mp_n_root'
+   );
+
+# The global variable where gather_functions() puts all it's findings in.
+my @dependency_list = ();
+
+# Uniquefy an array.
+sub uniq {
```

Comment:
`uniq` could be  supplied by [List::MoreUtils](https://metacpan.org/pod/List::MoreUtils) like `any`

Reply:
Ah, thanks!
But I'm pretty sure there are more, how shall I put it, gems like that and also worse ones in it. It is in urgent need for a Perl-expert to take a closer look.